src/FluentMigrator.Abstractions/Builders/IColumnOptionSyntax.cs

```diff
@@ -111,6 +111,14 @@ public interface IColumnOptionSyntax<TNext,TNextFk> : IFluentSyntax
         /// <returns>The next step</returns>
         TNext NotNullable();
 
+        /// <summary>
+        /// Defines the column as a computed column
+        /// </summary>
+        /// <param name="expression">The expression to calculate</param>
+        /// <param name="stored">Whether the computed column is virtual or stored</param>
+        /// <returns>The next step</returns>
+        TNext AsExpression(string expression, bool stored = false);
```

Comment:
Hello, just my two cents, but I think this naming may confuse people, because of C# Expression trees for example, while it is dedicated to computed columns. Why not name it simply `AsComputed(string expression, bool stored = false)` ?

Reply:
Makes sense. Also it’s now treated as an option, so like Identity or Indexed, maybe it should be called Computed?

Reply:
@PhenX Good question and your feedback serves as a "Keeper of the Flame."

I originally suggested AsExpression because the type is inferred by almost all database systems based on an expression's final type. In addition, I was thinking at some point specifying expressions from a POCO (similar to how third party package EasyMigrator works), so you *could* one day pass in an expression tree.

Incorporating expressions at some point in the future would also be a nice way to specify partition formulas. For a long time I didn't see a ton of value in the EasyMigrator approach but with features I want to do in the future like truly idempotent, offline Migrations and ability to test Migrations, having a current representation of what the schema should be described as entities is quite nice.

How confusing do you think it is? I guess the weird part is nobody in the sql world would call it an expression column.

Reply:
> the weird part is nobody in the sql world would call it an expression column

I think that’s important. I expect one of the goals is to make the library easy to use. If you were a user and were attempting to make a computed / calculated / virtual / generated column, what name would you look for?

Expression certainly still makes sense for the argument name even if it’s not the name of the method. 

> I originally suggested AsExpression because the type is inferred by almost all database systems based on an expression's final type

My research indicated that only SQL Server was able to infer the data type; all other databases require an explicit data type. Hence moving the method to an option.

Reply:
Ok. I'm out voted 2 to 1. AsComputed

Reply:
lol I’m okay with whatever. Just my opinion. 

Do you think it should be AsComputed or just Computed?  We don’t have AsIdentity which is also a generated column…

Reply:
I find `Computed` may be a better choice especially if we may still need to specify the column type with the "as" methods

Reply:
Good point. Computed() reads better.

We could also add an `IColumnTypeSyntax.AsComputed<TDbType>(string expression, bool stored = false)` that wraps the long-form expression, e.g.

```c#
.AsComputed<int>("USER_ID()", true)
```

would be short-form for

```c#
.AsInt32().Computed("USER_ID(), true)
```

The relative advantage of the short-form is API discoverability.

`ExpressionBuilderWithColumnTypesBase `would then be something like:

```c#
        /// <inheritdoc />
        public TNext AsComputed<TDbType>(string expression, bool stored = false)
        {
            Column.Type = GetDbType<TDbType>();
            Column.ExpressionValue = expression;
            Column.ExpressionStored = stored;
            return (TNext)(object)this;
        }

        private DbType GetDbType<TDbType>()
        {
             var typeCode = Type.GetTypeCode(typeof(TDbType);
             switch (typeCode)
             {
                  case TypeCode.String: return DbType.String;
                  case TypeCode.Int32: return DbType.Int32;
                  case TypeCode.Int64: return DbType.Int64;
                  case TypeCode.Decimal: return DbType.Decimal;
                  // etc
                  default:
                       throw new InvalidOperationException("TDbType type code {typeCode} does not map to a FluentMigrator DbType.");
             }
        }
```

Reply:
Thinking about my prototype above a bit more after lunch break - for Decimal and String, this interface is non-ideal since we wouldnt know the length of the string or whether it's an ANSI or UTF string, fixed or variable, or in the case of decimal, the scale and precision.

Reply:
Okay, I'll work on the 'long-form' first (when I have time) and then we can always add the short-form as subsequent PR if desired.